capstone/src/main/resources/static/user.js

```diff
@@ -0,0 +1,375 @@
+  }
+}
+
+/* get and store the user type (individual or organization). With this we determine whether
+this user exists in our database or not */
+function getUserType() {
+    /* Since there is no way to know beforehand whether the user is an organization
+    or an individual, we have to do two fetches to check the organization entities and
+    the user entities */
+    fetch('get-organization').then(response => response.json()).then((data) => {
+      if (data.length != 0) {
+          displayNavToggle("individual-nav", "org-nav");
+          setupAndStore(data[0], true);
+      } else {
+        displayNavToggle("org-nav", "individual-nav");
+        fetch('get-individual').then(response => response.json()).then((newData) => {
+          if (newData.length != 0) {
+            // display information
+            setupAndStore(newData[0]);
+          } else {
+            // user does not exist at all, display message to them to submit a profile
+            displayMain(false);
+            displayForm("individual", true);
+          }
+        });
+      }
+  });
+}
+
+function displayNavToggle(hide, display) {
+  document.getElementById(display).style.display="block";
+  document.getElementById(hide).style.display="none";
+}
+
+/** helper function to store information and set up display */
+function setupAndStore(data) {
+  sessionStorage.setItem("user-type", data.userType);
+  sessionStorage.setItem("university", data.university);
+  displayMain(true);
+}
+
+/* creates and populates the user profile */
+function createProfile(data, fillForm, isOrganization) {
+    const emailContainer = document.getElementById("email");
+    const pElementEmail = document.createElement('p');
+    pElementEmail.innerText = data.email;
+    emailContainer.appendChild(pElementEmail);
+
+    const universityContainer = document.getElementById("university");
+    const pElementUniversity = document.createElement('p');
+    pElementUniversity.innerText = data.university;
+    universityContainer.appendChild(pElementUniversity);
+
+    // addresses the different fields for each user type
+    if (isOrganization) {
+      createOrgProfile(data, fillForm);
+    } else {
+      createIndividualProfile(data, fillForm);
+    }
+
+}
+
+/* populate individual specific fields of the profile */
+function createIndividualProfile(data, fillForm) {
+  const nameContainer = document.getElementById("name");
+  const pElementName = document.createElement('h1');
+  pElementName.innerText = data.firstName + " " + data.lastName;
+  nameContainer.appendChild(pElementName);
+
+  // hide fields that pertain to organizations only
+  document.getElementById("description").style.display = "none";
+
+  if (fillForm) {
+    // prefill form
+    document.getElementById("ind-firstname").value = data.firstName;
+    document.getElementById("ind-lastname").value = data.lastName;
+    document.getElementById("university-form-display").innerText = data.university;
```

Comment:
oh shoot in my pr, I hide this element bc it was creating white space in the form. I had it "revealed" in this function but in script.js: `document.getElementById("university-form-display").display = 'block';`

Reply:
Ohh yeah, it basically is a toggle between this university p element and the university input element depending on whether the user had entered a university before.